app/lib/link_checker/uri_checker/http_checker.rb

```diff
@@ -158,6 +158,8 @@ def check_request
 
       add_problem(SlowResponse.new(from_redirect: from_redirect?)) if response_time > RESPONSE_TIME_WARNING
 
+      return if response.nil?
```

Comment:
Looking good so far!

I'd be tempted to move this onto line 157, to minimise the number of places the method can return from:

```ruby
      return response if report.has_errors? || response.nil?
```

Both have the effect of returning `nil`, though as you've pointed out, the return value isn't used.

Also, is it possible to write a test for this?